### ISSUE TITLE
SEACAS: Exodus missing optional dependency on Pnetcdf

### ISSUE THREAD

**Issue by jhux2** [trilinos/Trilinos#6078]
### Description
The ~~Nemesis~~ Exodus library is missing an optional dependency specification on Pnetcdf.  As a result, I see compilation failures like

```
[1/781] Linking CXX executable packages/seacas/libraries/nemesis/SEACASNemesis_ne_ctest_wrap.exe
FAILED: packages/seacas/libraries/nemesis/SEACASNemesis_ne_ctest_wrap.exe 
: && /ascldap/users/projects/ppc64le-pwr9-nvidia/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/bin/mpicxx  -pedantic -Wall -Wno-long-long -Wwrite-strings -Wshadow -Woverloaded-virtual   -Wno-deprecated-declarations -std=c++11 -O3 -DNDEBUG   packages/seacas/libraries/nemesis/CMakeFiles/SEACASNemesis_ne_ctest_wrap.dir/ne_ctest_wrap.c.o  -o packages/seacas/libraries/nemesis/SEACASNemesis_ne_ctest_wrap.exe  packages/seacas/libraries/nemesis/libnemesis.a packages/seacas/libraries/exodus/libexodus.a /ascldap/users/projects/ppc64le-pwr9-nvidia/netcdf-exo/4.6.1/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/lib/libnetcdf.so /ascldap/users/projects/ppc64le-pwr9-nvidia/hdf5/1.8.20/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/lib/libhdf5.so /ascldap/users/projects/ppc64le/zlib/1.2.8/lib/libz.a /usr/lib64/libdl.so /usr/lib64/libm.so && :
/home/projects/ppc64le/binutils/2.30.0/lib/../powerpc64le-unknown-linux-gnu/bin/ld: packages/seacas/libraries/exodus/libexodus.a(ex_utils.c.o): undefined reference to symbol 'ncmpi_inq_libvers'
/home/projects/ppc64le-pwr9-nvidia/pnetcdf-exo/1.9.0/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/lib/libpnetcdf.so.1: error adding symbols: DSO missing from command line
```

This patch fixes the problem for me:

```
diff --git a/packages/seacas/libraries/exodus/cmake/Dependencies.cmake b/packages/seacas/libraries/exodus/cmake/Dependencies.cmake
index e826e82..97a7ded 100644
--- a/packages/seacas/libraries/exodus/cmake/Dependencies.cmake
+++ b/packages/seacas/libraries/exodus/cmake/Dependencies.cmake
@@ -1,6 +1,6 @@
 TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
   LIB_REQUIRED_TPLS Netcdf
-  LIB_OPTIONAL_TPLS Pthread HDF5 MPI
+  LIB_OPTIONAL_TPLS Pthread HDF5 MPI Pnetcdf
   LIB_OPTIONAL_DEP_TPLS Pthread HDF5 MPI
   TEST_REQUIRED_TPLS Netcdf
   TEST_OPTIONAL_TPLS Pthread

```

### Steps to Reproduce
1. SHA1:  Trilinos dev VOTD

on waterman:

```
module load devpack/20180517/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88
module swap cmake/3.6.2 cmake/3.12.3
module load superlu/4.3.0/gcc/7.2.0
```

<details>
<summary>cmake configure</summary>

```bash
#!/bin/bash

# Where trilinos was cloned
trilinos_source_dir=${TRILINOS_DIR}

# Where trilinos will be installed
trilinos_install_dir=/ascldap/users/jhu/exawind/install-trilinos-waterman-relwithdebinfo

EXTRA_ARGS=$@

CXX_WARNINGS="-Wno-deprecated-declarations"

# Cleanup old cache before we configure
# Note:  This does not remove files produced by make.  Use "make clean" for this.
find . -name "CMakeFiles" -exec rm -rf {} \;
rm -f CMakeCache.txt

ARGS=(
  -GNinja
  -D CMAKE_INSTALL_PREFIX:PATH="${trilinos_install_dir}"
  -D CMAKE_BUILD_TYPE:STRING=RELEASE
  -D CMAKE_CXX_FLAGS:STRING="${CXX_WARNINGS}"
  -D Trilinos_ENABLE_CXX11:BOOL=ON
  -D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF
  -D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON
  -D Trilinos_ENABLE_TESTS:BOOL=OFF
  -D Trilinos_ALLOW_NO_PACKAGES:BOOL=OFF
  -D Trilinos_ASSERT_MISSING_PACKAGES:BOOL=OFF
  -D Trilinos_ENABLE_Amesos:BOOL=OFF
  -D Trilinos_ENABLE_Amesos2:BOOL=ON
  -D Amesos2_ENABLE_SuperLU:BOOL=ON
  -D Amesos2_ENABLE_KLU2:BOOL=OFF
  -D Trilinos_ENABLE_SuperLU5_API:BOOL=OFF
  -D Trilinos_ENABLE_AztecOO:BOOL=OFF
  -D Trilinos_ENABLE_Belos:BOOL=ON
  -D Trilinos_ENABLE_Epetra:BOOL=OFF
  -D Trilinos_ENABLE_EpetraExt:BOOL=OFF
  -D Trilinos_ENABLE_Gtest:BOOL=ON
  -D Trilinos_ENABLE_Ifpack:BOOL=OFF
  -D Trilinos_ENABLE_Ifpack2:BOOL=ON
  -D Trilinos_ENABLE_ML:BOOL=OFF
  -D Trilinos_ENABLE_MueLu:BOOL=ON
  -D Trilinos_ENABLE_SEACASExodus:BOOL=ON
  -D Trilinos_ENABLE_SEACASEpu:BOOL=ON
  -D Trilinos_ENABLE_SEACASExodiff:BOOL=ON
  -D Trilinos_ENABLE_SEACASNemspread:BOOL=ON
  -D Trilinos_ENABLE_SEACASNemslice:BOOL=ON
  -D Trilinos_ENABLE_SEACASIoss:BOOL=ON
  -D Trilinos_ENABLE_STK:BOOL=ON
  -D Trilinos_ENABLE_STKIO:BOOL=ON
  -D Trilinos_ENABLE_Tpetra:BOOL=ON
  -D   Tpetra_INST_INT_LONG:BOOL=ON
  -D   Tpetra_INST_DOUBLE:BOOL=ON
  -D   Tpetra_INST_COMPLEX_DOUBLE:BOOL=OFF
  -D Trilinos_ENABLE_Zoltan:BOOL=ON
  -D Trilinos_ENABLE_Zoltan2:BOOL=ON
  -D   BoostLib_INCLUDE_DIRS:PATH="${BOOST_ROOT}/include"
  -D   BoostLib_LIBRARY_DIRS:PATH="${BOOST_ROOT}/lib"
  -D TPL_ENABLE_HDF5:BOOL=ON
  -D   HDF5_DIR:PATH="${SEMS_HDF5_ROOT}"
  -D TPL_ENABLE_MPI:BOOL=ON
  -D   MPI_BASE_DIR:PATH="${MPI_ROOT}"
  -D TPL_ENABLE_Netcdf:BOOL=ON
  -D   Netcdf_INCLUDE_DIRS:PATH="${NETCDF_ROOT}/include"
  -D   Netcdf_LIBRARY_DIRS:PATH="${NETCDF_ROOT}/lib"
  -D TPL_Netcdf_PARALLEL:BOOL=ON
  -D TPL_ENABLE_Pnetcdf:BOOL=ON
  -D   Pnetcdf_INCLUDE_DIRS:PATH="${PNETCDF_ROOT}/include"
  -D   Pnetcdf_LIBRARY_DIRS:PATH="${PNETCDF_ROOT}/lib"
  -D TPL_ENABLE_SuperLU:BOOL=ON
  -D   SuperLU_INCLUDE_DIRS:PATH="${SUPERLU_ROOT}/include"
  -D   SuperLU_LIBRARY_DIRS:PATH="${SUPERLU_ROOT}/lib"
  -D TPL_ENABLE_BLAS:BOOL=ON
  -D   BLAS_LIBRARY_DIRS:PATH="${BLAS_ROOT}/lib"
  -D TPL_ENABLE_LAPACK:BOOL=ON
  -D   LAPACK_LIBRARY_DIRS:PATH="${LAPACK_ROOT}/lib"
)

cmake "${ARGS[@]}" $EXTRA_ARGS $trilinos_source_dir
```

</details>

**Comment by jhux2** [comment]
@trilinos/seacas

**Comment by gsjaardema** [comment]
Thanks.  I have fixed in the SEACAS github repository and will snapshot into Trilinos this week.

**Comment by gsjaardema** [comment]
This should be fixed now via #6902

**Comment by jhux2** [comment]
> This should be fixed now

Thank you!
